net/src/ipc/transport_wss/mod.rs

```diff
@@ -85,9 +85,9 @@ impl<T: Read + Write + std::fmt::Debug> Transport for TransportWss<T> {
         let host_port = format!(
             "{}:{}",
             uri.host_str()
-                .ok_or(TransportError("bad connect host".into()))?,
+                .ok_or_else(|| TransportError("bad connect host".into()))?,
             uri.port()
-                .ok_or(TransportError("bad connect port".into()))?,
+                .ok_or_else(|| TransportError("bad connect port".into()))?,
```

Comment:
Why do you consider this change an improvement?

Reply:
Because the error creation and string conversion is done lazily this way

Reply:
But that doesn't make any difference in this line, or does it? So in what case would then use `.ok_or`? I would use it when the creation of the alternative has no side-effects and is basically just calling a constructor, like in this example.

Reply:
You can use `ok_or` if you have a static value that doesn't need evaluation at runtime